src/test/unit/althold_unittest.cc

```diff
@@ -0,0 +1,157 @@
+#include "unittest_macros.h"
+#include "gtest/gtest.h"
+
+extern "C" {
+
+    #include "pg/pg.h"
+    #include "pg/pg_ids.h"
+
+    #include "sensors/acceleration.h"
+    FAST_DATA_ZERO_INIT acc_t acc;
+    PG_REGISTER(accelerometerConfig_t, accelerometerConfig, PG_ACCELEROMETER_CONFIG, 0);
+
+    #include "build/debug.h"
+    int16_t debug[DEBUG16_VALUE_COUNT];
+    uint8_t debugMode;
+
+    #include "fc/runtime_config.h"
+    uint8_t armingFlags = 0;
+    uint8_t stateFlags = 0;
+    uint16_t flightModeFlags = 0;
+
+    #include "flight/imu.h"
+    void imuTransformVectorBodyToEarth(t_fp_vector * v)
+    {
+        (void)v;
+    }
+
+    #include "drivers/time.h"
+    uint32_t millisRW;
+    uint32_t millis() {
+        return millisRW;
+    }
+
+    #include "sensors/barometer.h"
+    baro_t baro;
+
+    #include "flight/position.h"
+    float getAltitude(void) { return 0.0f; }
+    PG_REGISTER(positionConfig_t, positionConfig, PG_POSITION, 0);
+
+    #include "fc/rc_controls.h"
+    float rcCommand[4];
+
+    #include "rx/rx.h"
+    void parseRcChannels(const char *input, rxConfig_t *rxConfig)
+    {
+        UNUSED(input);
+        UNUSED(rxConfig);
+    }
+
+    // other
+    float getRcDeflection(int)
+    {
+        return 0;
+    }
+
+    bool baroIsCalibrated(void)
+    {
+        return 1;
+    }
+
+    bool sensors(uint32_t)
+    {
+        return 1;
+    }
+
+    bool failsafeIsActive(void)
+    {
+        return 0;
+    }
+
+    #include "flight/alt_hold.h"
+
+    extern altHoldState_s altHoldState;
+    void altHoldReset();
+    void altHoldProcessTransitions();
+    void altHoldInit();
+    void altHoldUpdate();
+    void pt2FilterInit(pt2Filter_t *altHoldDeltaLpf, float) {
+        UNUSED(altHoldDeltaLpf);
+    }
+    float pt2FilterGain(float, float) {
+        return 0.0f;
+    }
+    float pt2FilterApply(pt2Filter_t *altHoldDeltaLpf, float) {
+        UNUSED(altHoldDeltaLpf);
+        return 0.0f;
+    }
```

Comment:
You don't need to define all these stub functions. We just need to include the .c files in the unittest makefile

Reply:
I tidies this code up as best I could, but the lines `extern altHoldState_s altHoldState;... ` needed to be in the opening area.  Happy to move them so long as it still works, but the seem to beed to be where they are.  Everything else is tidied up.  I don't understand these tests.  If someone could PR to make the test useful, or give exact code that would make it work, that would be awesome.  I can't do it, I don't know how :-(